packages/liveblocks-react-tiptap/src/LiveblocksExtension.ts

```diff
@@ -0,0 +1,259 @@
+import {
+  useClient,
+  useCommentsErrorListener,
+  useRoom,
+} from "@liveblocks/react";
+import {
+  CreateThreadError,
+  getUmbrellaStoreForClient,
+} from "@liveblocks/react/_private";
+import { LiveblocksYjsProvider } from "@liveblocks/yjs";
+import type { AnyExtension } from "@tiptap/core";
+import { Extension, getMarkType } from "@tiptap/core";
+import Collaboration from "@tiptap/extension-collaboration";
+import CollaborationCursor from "@tiptap/extension-collaboration-cursor";
+import type { Mark as PMMark } from "@tiptap/pm/model";
+import { useCallback } from "react";
+import { Doc } from "yjs";
+
+import { CommentsExtension } from "./comments/CommentsExtension";
+import { MentionExtension } from "./mentions/MentionExtension";
+import { LIVEBLOCKS_COMMENT_MARK_TYPE } from "./types";
+
+const providersMap = new Map<
+  string,
+  LiveblocksYjsProvider<any, any, any, any, any>
+>();
+
+const docMap = new Map<string, Doc>();
+
+type LiveblocksExtensionOptions = {
+  field?: string;
+  comments: boolean; // | CommentsConfiguration
+  mentions: boolean; // | MentionsConfiguration
+};
+
+const LiveblocksCollab = Collaboration.extend({
+  // Override the onCreate method to warn users about potential misconfigurations
+  onCreate() {
+    if (
+      !this.editor.extensionManager.extensions.find((e) => e.name === "doc")
+    ) {
+      console.warn(
+        "[Liveblocks] The tiptap document extension is required for Liveblocks collaboration. Please add it or use Tiptap StarterKit extension."
+      );
+    }
+    if (
+      !this.editor.extensionManager.extensions.find(
+        (e) => e.name === "paragraph"
+      )
+    ) {
+      console.warn(
+        "[Liveblocks] The tiptap paragraph extension is required for Liveblocks collaboration. Please add it or use Tiptap StarterKit extension."
+      );
+    }
+
+    if (
+      !this.editor.extensionManager.extensions.find((e) => e.name === "text")
+    ) {
+      console.warn(
+        "[Liveblocks] The tiptap text extension is required for Liveblocks collaboration. Please add it or use Tiptap StarterKit extension."
+      );
+    }
+    if (
+      this.editor.extensionManager.extensions.find((e) => e.name === "history")
+    ) {
+      console.warn(
+        "[Liveblocks] The history extension is enabled, Liveblocks extension provides its own. Please remove or disable the History plugin to prevent unwanted conflicts."
+      );
+    }
+  },
+});
+
+export const useLiveblocksExtension = (): Extension => {
+  const room = useRoom();
+
+  // TODO: we don't need these things if comments isn't turned on...
+  // TODO: we don't have a reference to the editor here, need to figure this out
+  useCommentsErrorListener((error) => {
+    // If thread creation fails, we remove the thread id from the associated nodes and unwrap the nodes if they are no longer associated with any threads
+    if (error instanceof CreateThreadError) {
```

Comment:
NOTE: we have a ticket for this here: https://linear.app/liveblocks/issue/LB-1356/listen-for-comment-creation-fail-events

but didn't want to block the beta release